webapp/api/github.py

```diff
@@ -0,0 +1,139 @@
+import flask
+from urllib.parse import urlencode
+from webapp import api
+from requests.exceptions import RequestException
+
+
+class GithubAPIError(RequestException):
+    pass
+
+
+class GithubAPI:
+    """
+    Provides authentication for GitHub users. Helper methods are also provided
+    for checking organization access and getting user data from the Github API.
+    """
+
+    BASE_URL = "https://api.github.com/"
+    AUTH_URL = "https://github.com/login/oauth/"
+
+    def __init__(
+        self, client_id, client_secret, session=api.requests.Session()
+    ):
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.session = session
+        self.session.headers["Accept"] = "application/json"
+
+    def authorize(self, scope=None, state=None):
```

Comment:
I don't feel this method belongs on this class. This class should be about retrieving data from the GitHub API, not redirecting the user. Redirecting the user is a view-level concern.

I think all the logic in this function should instead be moved directly to the `github_auth` view function.

Reply:
I agree, but is it fine to have a method that only build and return the authorization URL to GitHub and then use that URL in the view to do the redirect?